### PR TITLE
docs: Add initial draft of a migration guide

### PR DESCRIPTION
**Pull request by alcarney**
I _think_ this covers all of the planned breaking changes between the two versions

### CODE REVIEW

File: docs/lsp/howto/migrate-to-v1.rst [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+   Also the details in this guide may change over time.
+
+This guide covers the breaking changes between the ``v0.x`` and ``v1.x`` versions of the language server and how to adapt to them.
+
+.. highlight:: console
+
+Installation Changes
+--------------------
+
+Previously, you would have had to install ``esbonio`` as a development dependency for every project you wished to use it in.
```
Comment (polyzen) [inline docs/lsp/howto/migrate-to-v1.rst:18]: Has also been working fine when installed globally (to `/usr/lib/python3.xx/site-packages`), as done in https://archlinux.org/packages/extra/any/esbonio/.

Comment (polyzen) [reply docs/lsp/howto/migrate-to-v1.rst:18]: Oh, I suppose this was a limitation of the VS Code extension.

File: docs/lsp/howto/migrate-to-v1.rst [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+| - ``esbonio.sphinx.buildDir``           |                                                 |             |
+| - ``esbonio.sphinx.builderName``        |                                                 |             |
+| - ``esbonio.sphinx.confDir``            |                                                 |             |
+| - ``esbonio.sphinx.doctreeDir``         |                                                 |             |
+| - ``esbonio.sphinx.forceFullBuild``     |                                                 |             |
+| - ``esbonio.sphinx.keepGoing``          |                                                 |             |
+| - ``esbonio.sphinx.makeMode``           |                                                 |             |
+| - ``esbonio.sphinx.numJobs``            |                                                 |             |
+| - ``esbonio.sphinx.quiet``              |                                                 |             |
+| - ``esbonio.sphinx.silent``             |                                                 |             |
```
Comment (polyzen) [inline docs/lsp/howto/migrate-to-v1.rst:61]: Not sure if you're planning on re-implementing `silent`, but it is handy for de-duplicating diagnostics when also using rstcheck (rstcheck has Sphinx support, but also works without Sphinx).

File: docs/lsp/howto/migrate-to-v1.rst (hunk 1 above)
Comment (alcarney) [reply docs/lsp/howto/migrate-to-v1.rst:18]: Thanks for the feedback!

> Has also been working fine when installed globally 

The guide should probably have said something like the "recommended approach was..."

The key thing is that the `0.x` version needs to be installed alongside Sphinx and all of your docs dependencies in order to work properly. In the new version this restriction is going to go away :)

File: docs/lsp/howto/migrate-to-v1.rst (hunk 2 above)
Comment (alcarney) [reply docs/lsp/howto/migrate-to-v1.rst:61]: Each of the removed config options should correspond with a cli option you can pass to `sphinx-build`. For example, there is the [`-Q`](https://www.sphinx-doc.org/en/master/man/sphinx-build.html#cmdoption-sphinx-build-Q) flag which should do the equivalent to what `esbonio.sphinx.silent` did. (Feel free to open an issue if it does not!)

I should probably update the guide pointing out the equivalent `sphinx-build` flag for each of the removed options

**Pull request merged**